cpp/include/cudf/transform.hpp

```diff
@@ -142,5 +142,40 @@ std::unique_ptr<column> mask_to_bools(
   size_type end_bit,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
 
+/**
+ * @brief Returns an approximate cumulative size in bits of all columns in the `table_view` for
+ * each row.
+ *
+ * This function counts bits instead of bytes to account for the null mask which only has one
+ * bit per row.
+ *
+ * Each row in the returned column is the sum of the per-row size for each column in
+ * the table.
+ *
+ * In some cases, this is an inexact approximation. Specifically, with
+ * lists or strings, the cost of a row includes 32 bits for a single offset. However, two
+ * offsets is required to represent an entire row.  But this presents a problem, because to
+ * represent 2 rows, you need 3 offsets.  3 rows 4 offsets, etc.  Therefore it would not
+ * be accurate to say each row of a string column costs 2 offsets because summing multiple row
+ * sizes would give you a number too large. It is up to the caller to understand the schema
+ * of the input column to be able to calculate the small additional overhead of the
+ * terminating offset for any group of rows being considered.
```

Comment:
I think this could be simpler and clearer. It's just that there are N+1 offsets for N rows (because we store an end offset to be able to calculate lengths).

```suggestion
 * In some cases, this is an inexact approximation. Specifically, columns of lists and strings 
 * require N+1 offsets to represent N rows. It is up to the caller to calculate the small 
 * additional overhead of the terminating offset for any group of rows being considered.
```